### PR TITLE
Bump storybook version to v3 and webpack to v2

### PR DESCRIPTION
**Pull request by majapw**
Hooray!

There is one small issue with @storybook/info-addon where we get a proptype warning in the console (it is on the way to being fixed - https://github.com/storybooks/storybook/pull/1374). Other than that, the storybook, build script, and tests all seem to work as intended.

@ljharb can you take another look?

### CODE REVIEW

File: .storybook/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "rules": {
+    "import/no-extraneous-dependencies": [2, {
+      "devDependencies": true
+    }],
+    "react/jsx-filename-extension": 2,
```
Comment (ljharb) [inline .storybook/.eslintrc]: why can't our jsx files be .jsx for storybook?

File: .storybook/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    "global-require": 2,
+    "comma-dangle": 2
```
Comment (ljharb) [inline .storybook/.eslintrc]: also, por que?

File: .storybook/addons.js [hunk 3]
```diff
@@ -1,2 +1,4 @@
-import '@kadira/storybook/addons';
-import '@kadira/storybook-addon-options/register';
+/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
```
Comment (ljharb) [inline .storybook/addons.js]: why is this override needed?

File: .storybook/.eslintrc (hunk 1 above)
Comment (majapw) [inline .storybook/.eslintrc]: i think they might have to be js due to storybook's internal config?

File: .storybook/.eslintrc (hunk 2 above)
Comment (majapw) [inline .storybook/.eslintrc]: mostly to get my webpack config to stop yelling at me in sublime but also i guess maybe not necessary.

also not working. :/

File: .storybook/addons.js (hunk 3 above)
Comment (majapw) [inline .storybook/addons.js]: This is the default storybook config as per the slow start guide. The reason is that these modules are included as part of the `@storybook/react` package and not explicitly defined in our package.json

File: stories/DateRangePicker_calendar.js [hunk 4]
```diff
@@ -136,12 +136,6 @@ storiesOf('DRP - Calendar Props', module)
       autoFocus
     />
   ))
-  .addWithInfo('with keyboard shorcuts panel hidden', () => (
```
Comment (majapw) [inline stories/DateRangePicker_calendar.js]: This was a duplicate story which used to just be ignored but now errors out.

File: .storybook/.eslintrc (hunk 1 above)
Comment (ljharb) [inline .storybook/.eslintrc]: k

File: .storybook/addons.js (hunk 3 above)
Comment (ljharb) [inline .storybook/addons.js]: sure, but `no-unresolved` still should be validating that it's importable, no?

File: .storybook/webpack.config.js [hunk 5]
```diff
@@ -2,19 +2,35 @@ const path = require('path');
...
-    loaders: [
+    rules: [
       {
-        test:   /\.scss$/,
-        loaders: ['style', 'raw', 'sass'],
-        include: path.resolve(__dirname, '../css/')
+        test: /\.scss$/,
+        use: ['style-loader', 'raw-loader', 'sass-loader'],
+        include: [
+          path.resolve(__dirname, './'),
```
Comment (lencioni) [inline .storybook/webpack.config.js]: Is this intentional? If you have this line, do you need the next line?

Comment (majapw) [inline .storybook/webpack.config.js]: Yes, so this is looking in the `/.storybook` config folder, which contains the scss for the storybook whereas the next line looks for the `react-dates` scss files, which are located in the `/css` folder

**Pull request merged**